src/apps/viewer/extensions/cellpack/model.ts

```diff
@@ -34,40 +35,54 @@ function getCellPackModelUrl(fileName: string, baseUrl: string) {
     return `${baseUrl}/results/${fileName}`
 }
 
-async function getModel(id: string, baseUrl: string) {
+async function getModel(id: string, source:IngredientSource, baseUrl: string) {
     let model: Model;
+    const mid = source.model? parseInt(source.model) : 0;
     if (id.match(/^[1-9][a-zA-Z0-9]{3,3}$/i)) {
         // return
         const cif = await getFromPdb(id)
-        model = (await trajectoryFromMmCIF(cif).run())[0]
+        model = (await trajectoryFromMmCIF(cif).run())[mid]
     } else {
         const pdb = await getFromCellPackDB(id, baseUrl)
-        model = (await trajectoryFromPDB(pdb).run())[0]
+        model = (await trajectoryFromPDB(pdb).run())[mid]
     }
     return model
 }
 
-async function getStructure(model: Model, props: { assembly?: string } = {}) {
+async function getStructure(model: Model, source:IngredientSource, props: { assembly?: string } = {}) {
     let structure = Structure.ofModel(model)
     const { assembly } = props
 
     if (assembly) {
         structure = await StructureSymmetry.buildAssembly(structure, assembly).run()
     }
-
-    const query = MS.struct.modifier.union([
-        MS.struct.generator.atomGroups({
-            'entity-test': MS.core.rel.eq([MS.ammp('entityType'), 'polymer'])
-        })
-    ])
-    const compiled = compile<StructureSelection>(query)
-    const result = compiled(new QueryContext(structure))
-    structure = StructureSelection.unionStructure(result)
-
+    if (source.selection){
+        //use NGL selection string or :A or :B etc...
+        const asymIds:string[] = source.selection.replace(" :","").split(" or")
+        console.log(asymIds)
+        const query = MS.struct.modifier.union([
+            MS.struct.generator.atomGroups({
+                'chain-test': MS.core.set.has([MS.set(...asymIds), MS.ammp('label_asym_id')])
```

Comment:
I think you mean `auth_asym_id` here. Also you should add
```
'entity-test': MS.core.rel.eq([MS.ammp('entityType'), 'polymer'])
```
as well because `auth_asym_id`s can contain ligands as well or do you want to include them?